site/src/github.rs

```diff
@@ -33,8 +33,19 @@ pub async fn unroll_rollup(
         let display = truncate.then(|| s.split_at(10).0).unwrap_or(s);
         format!("[{display}](https://github.com/rust-lang-ci/rust/commit/{s})")
     };
-    let mapping = enqueue_unrolled_try_builds(ci_client, rollup_merges, previous_master)
-        .await?
+
+    // Sort rolled up commits by their PR number in ascending order, so that they have the
+    // same ordering as in the rollup PR description.
+    let mut unrolled_builds: Vec<UnrolledCommit> =
+        enqueue_unrolled_try_builds(ci_client, rollup_merges, previous_master).await?;
+    unrolled_builds.sort_unstable_by(|a, b| {
+        a.original_pr_number
+            .parse::<u64>()
+            .and_then(|a| b.original_pr_number.parse::<u64>().map(|b| a.cmp(&b)))
+            .unwrap_or_else(|_| a.original_pr_number.cmp(b.original_pr_number))
+    });
```

Comment:
If the strings don't parse correctly, the ordering will be weird anyways, so `sort_by_key` could be enough here ?

```suggestion
unrolled_builds.sort_by_key(|a| a.original_pr_number.parse::<u64>().ok());
```

(or `sort_by_cached_key`)

Reply:
Good idea, I changedi t.